File: Day08/baekjoon-9996.py
num=int(input())
target=[]
for i in range(num+1):
    s=input().strip()
    target.append(s)

# target[0] 내 저장된 문자열에서 *을 중심으로 앞, 뒤를 나눈다.

# 1. 먼저 *을 찾는다.
star_idx=target[0].find('*')
front = target[0][:star_idx]
back = target[0][star_idx+1:]

for i in range(1,num+1):
    if len(target[i]) >= len(front) + len(back):
        if target[i][:len(front)] == front and target[i][len(target[i])-len(back):] == back:
            print("DA")
        else:
            print("NE")
    else:
        print("NE")

File: Day08/baekjoon-9996.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>
using namespace std;

int main(void)
{
    int num = 0;
    cin >> num;
    vector<string> target(num + 1, "");
    // target의 첫번째 인덱스에 패턴을 제공한다.
    for (int i = 0; i < num + 1; i++)
    {
        cin >> target[i];
    }
    int star_idx = target[0].find('*');
    string front = target[0].substr(0, star_idx);
    string back = target[0].substr(star_idx + 1);

    for (int i = 1; i < num + 1; i++)
    {
        if (target[i].length() >= front.length() + back.length())
        {
            if (target[i].substr(0, front.length()) == front && target[i].substr(target[i].length() - back.length()) == back)
            {
                cout << "DA" << '\n';
            }
            else
            {
                cout << "NE" << '\n';
            }
        }
        else
        {
            cout << "NE" << '\n';
        }
    }
    return 0;
}